doc/design/components.md

```diff
@@ -0,0 +1,252 @@
+
+As of the v0.1 release logs are being persisted on the datastore along with the VM.
+
+[Issues relating to container logging](https://github.com/vmware/vic/labels/component%2Fcontainer-logging)
+
+
+## Appliance
+### Appliance Base
+
+The appliance VM is bootstrapped from a PhotonOS based liveCD that embeds elements of VIC relevent to Virtual Container Host functions. The appliance is diskless _in essence_, but may well use a non-persistent disk to cache transient data such as docker images in-flight from [Docker Hub](https://hub.docker.com/) but not yet persisted to a datastore.
+
+The ISO used to bootstrap the appliance is generated from the following make targets - only the first must be specified as the others are pre-requisites:
+```
+make appliance appliance-staging iso-base
+```
+
+The intent behind having the appliance be diskless and bootstrap off an ISO each time is:
+* improved security - coupled with [the configuration mechanism](configuration.md#Configuration-persistence-mechanism) there's no way to make persistent changes without vSphere API access
+* robustness - a reboot will return the appliance to a known good state, assuming an administrator has not altered the configuration in the meantime
+* simplicity of update - using [vic-machine](#vic-machine) to update a VCH should be as simple as pointing the appliance VM at a new ISO version and rebooting, so long as there's no metadata migration needed; in that case the migration should be viable as a background task prior to VCH update.
+
+[Issues relating to appliance base](https://github.com/vmware/vic/labels/component%2Fappliance-base)
+
+
+### vicadmin
+
+This is more speculative than any of the other components at this point. We fully expect there to be a need for user level inspection/administration of a deployed Virtual Container Host, however we've not yet identified the functions this should provide.
+
+Current list of functions:
+* log collection
+
+Speculative list of functions (via docker-machine as a client?):
+* docker API user management
+* reconfigure operations (e.g. add --insecure-registry)
+
+[Issues relating to vicadmin](https://github.com/vmware/vic/labels/component%2Fvicadmin)
+
+
+### Docker API server
+
+This is the portion of a Virtual Container Host that provides a Docker API endpoint for users to interact with; it is also referred to as a 'personality'. The longer term design has multiple personalities running within a single VCH, such that the same endpoint can serve mixed API versions, and the same VCH can serve multiple API flavours.
+
+As of the v0.1 release this makes use of the [docker engine-api](https://github.com/docker/engine-api) project to ensure API compatibility with docker.
+
+The following is commentary about why the docker API server is not actual Docker. It's not what would usually be found in design doc, but is a question that comes up enough that it becomes core to understanding the design.
```

Comment:
need to add an "a" -- "It's not what would usually be found in **a** design doc, but is a question that comes up enough that it becomes core to understanding the design."